keps/sig-network/20200518-fqdn-in-hostname.md

```diff
@@ -0,0 +1,510 @@
+know that this has succeeded?
+-->
+
+Giving users the ability to set the hostname field of the kernel to the FQDN of a Pod.
+
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+Giving users a way to configure or enforce this feature cluster-wide.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+This proposal gives users the ability to set a pod’s hostname to its FQDN. A new PodSpec field named `fqdnInHostname` 
+will be introduced, with type `*bool`.
+
+The values of `fqdnInHostname` are:
+* `nil` (default): The Linux kernel hostname field ([the nodename field of struct utsname](http://man7.org/linux/man-pages/man2/uname.2.html)) 
+of a pod will be set to its shortname. This is the current behavior.
+* `False`: Same as `nil`
+* `True`: The Linux kernel hostname field ([the nodename field of struct utsname](http://man7.org/linux/man-pages/man2/uname.2.html)) 
+of a pod will be set to its fully qualified domain name (FQDN). FQDN is determined as described [here](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service))
+
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+#### Story 1: User does not Configure Pod to have FQDN
+Assume we have a pod named `foo` in a namespace `bar`. The PodSpec `subdomain` is not set. This pod does not have FQDN, so the value of `fqdnInHostname` does not have an impact. The Pod spec for this example would be:
+
+```yaml
+# Pod spec
+apiVersion: v1
+kind: Pod
+metadata: {"name": "foo", "namespace": "bar"}
+spec:
+  ...
+
+```
+
+If we `exec` into the Pod:
+* `uname -n` returns `foo`
+* `hostname --fqdn` returns `foo`
+
+#### Story 2: User Configures Pod to have FQDN
+
+Assume we have a pod named `foo` in a namespace `bar`. The PodSpec `subdomain` is set to `test`. We also assume the cluster-domain is set to its default, i.e. `cluster.local`. The FQDN of this pod is defined as `foo.subdomain.bar.svc.cluster.local` (see details [here](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service)). The user does not set `fqdnInHostname`. The Pod spec for this example would be:
+
+```yaml
+# Pod spec
+apiVersion: v1
+kind: Pod
+metadata: {"name": "foo", "namespace": "bar"}
+spec:
+  ...
+  hostname: "bar"  # Optional for this example
+  subdomain: "test"
+```
+
+If we `exec` into the Pod:
+* `uname -n` returns `foo`
+* `hostname --fqdn` returns `foo.subdomain.bar.svc.cluster.local`
```

Comment:
Shouldn't this be `bar.test.bar.svc.cluster.local`?

Reply:
Fixed, thanks